### ISSUE TITLE
Unable to select correct chart with HelmRelease

### ISSUE THREAD

**Issue by Dispersia** [fluxcd/flux2#4677]
### Describe the bug

 I am trying to use the kong/ingress chart, however no matter how i set up the HelmRelease it seems to select kong/kong-kong, which isn't a valid chart. I am new to flux, so I may just be doing something wrong

### Steps to reproduce

HelmRepository of

```
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: HelmRepository
metadata:
  name: kong
  namespace: kong
spec:
  interval: 1h
  url: https://charts.konghq.com
```

and a helmrelease of

```
apiVersion: helm.toolkit.fluxcd.io/v2beta2
kind: HelmRelease
metadata:
  name: kong
  namespace: kong
spec:
  interval: 10m
  timeout: 5m
  chart:
    spec:
      chart: kong/ingress
      version: 0.12.0
      sourceRef:
        kind: HelmRepository
        name: kong
        namespace: kong
      interval: 5m
```

### Expected behavior

 It should select the kong/ingress chart correctly

### OS / Distro

Arch Linux

### Flux version

v2.2.3

### Flux check

➜  manifests-skillbound git:(main) flux check                                                                                                                                     
► checking prerequisites                                                                                                                                                          
✔ Kubernetes 1.28.7+k3s1 >=1.26.0-0                                                                                                                                               
► checking version in cluster                                                                                                                                                     
✔ distribution: flux-v2.2.3                                                                                                                                                       
✔ bootstrapped: true                                                                                                                                                              
► checking controllers                                                                                                                                                            
✔ helm-controller: deployment ready                                                                                                                                               
► ghcr.io/fluxcd/helm-controller:v0.37.4                                                                                                                                          
✔ kustomize-controller: deployment ready                                                                                                                                          
► ghcr.io/fluxcd/kustomize-controller:v1.2.2                                                                                                                                      
✔ notification-controller: deployment ready                                                                                                                                       
► ghcr.io/fluxcd/notification-controller:v1.2.4                                                                                                                                   
✔ source-controller: deployment ready                                                                                                                                             
► ghcr.io/fluxcd/source-controller:v1.2.4                                                                                                                                         
► checking crds                                                                                                                                                                   
✔ alerts.notification.toolkit.fluxcd.io/v1beta3                                                                                                                                   
✔ buckets.source.toolkit.fluxcd.io/v1beta2                                                                                                                                        
✔ gitrepositories.source.toolkit.fluxcd.io/v1                                                                                                                                     
✔ helmcharts.source.toolkit.fluxcd.io/v1beta2                                                                                                                                     
✔ helmreleases.helm.toolkit.fluxcd.io/v2beta2                                                                                                                                     
✔ helmrepositories.source.toolkit.fluxcd.io/v1beta2                                                                                                                               
✔ kustomizations.kustomize.toolkit.fluxcd.io/v1                                                                                                                                   
✔ ocirepositories.source.toolkit.fluxcd.io/v1beta2                                                                                                                                
✔ providers.notification.toolkit.fluxcd.io/v1beta3                                                                                                                                
✔ receivers.notification.toolkit.fluxcd.io/v1                                                                                                                                     
✔ all checks passed

### Git provider

GitHub

### Additional context

 When inspecting the logs i get:

```
apiVersion: helm.toolkit.fluxcd.io/v2beta2
kind: HelmRelease
metadata:
  creationTimestamp: "2024-03-19T15:39:17Z"
  finalizers:
  - finalizers.fluxcd.io
  generation: 3
  labels:
    kustomize.toolkit.fluxcd.io/name: apps
    kustomize.toolkit.fluxcd.io/namespace: flux-system
  name: kong
  namespace: kong
  resourceVersion: "18605"
  uid: f2047de1-3610-4b68-95f0-580b82ae31f7
spec:
  chart:
    spec:
      chart: kong/ingress
      interval: 5m
      reconcileStrategy: ChartVersion
      sourceRef:
        kind: HelmRepository
        name: kong
        namespace: kong
      version: 0.12.0
  interval: 10m
  timeout: 5m
status:
  conditions:
  - lastTransitionTime: "2024-03-19T16:02:27Z"
    message: Fulfilling prerequisites
    observedGeneration: 3
    reason: Progressing
    status: "True"
    type: Reconciling
  - lastTransitionTime: "2024-03-19T16:12:50Z"
    message: 'HelmChart ''kong/kong-kong'' is not ready: invalid chart reference:
      failed to get chart version for remote reference: no chart name found'
    observedGeneration: 3
    reason: HelmChartNotReady
    status: "False"
    type: Ready
  - lastTransitionTime: "2024-03-19T15:44:23Z"
    message: 'Helm install failed for release kong/kong with chart ingress@0.12.0:
      context deadline exceeded'
    observedGeneration: 1
    reason: InstallFailed
    status: "False"
    type: Released
  failures: 1
  helmChart: kong/kong-kong
  history:
  - chartName: ingress
    chartVersion: 0.12.0
    configDigest: sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
    digest: sha256:c632aa9474854aa955e6a0a2acfecf57ed5dee28d5082e899c9b8ae106149fd4
    firstDeployed: "2024-03-19T15:39:21Z"
    lastDeployed: "2024-03-19T15:39:21Z"
    name: kong
    namespace: kong
    status: failed
    version: 1
  installFailures: 1
  lastAttemptedConfigDigest: sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
  lastAttemptedGeneration: 1
  lastAttemptedReleaseAction: install
  lastAttemptedRevision: 0.12.0
  observedGeneration: 1
  storageNamespace: kong
```

it shows the spec/chart/spec/chart 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Dispersia** [comment]
Hello, thank you for the link to debug it, however the HelmRelease seems to be failing before the HelmChart is created (or at least, if i try to find it it doesnt exist so may be being cleaned up). Sorry, I am getting a bit lost, why is it pulling from the metadata name instead of the chart I specified? 

 But, I now changed it to
 
 ```
apiVersion: helm.toolkit.fluxcd.io/v2beta2
kind: HelmRelease
metadata:
  name: kong
  namespace: kong
spec:
  interval: 10m
  timeout: 5m
  chart:
    spec:
      chart: ingress
      version: 0.12.0
      sourceRef:
        kind: HelmRepository
        name: kong
        namespace: kong
      interval: 5m
```

but how would i ever be able to construct kong/ingress if it always tries to append the namespace? I think I'm getting confused on it saying helmChart: kong/kong-kong (or if i change the metadata.name to ingress, it becomes kong/kong-ingress), and not including my chart in the spec at all. Is that the HelmChart <namespace>/<chart-name> like actual k8s object it's trying to create, and not the helm it's trying to install? If so, that would make a bit more sense to me, but then feels like the above should be working
